src/Core/AdminConsole/OrganizationFeatures/Policies/PolicyValidators/OrganizationPolicyValidator.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Bit.Core.AdminConsole.Entities;
+using Bit.Core.AdminConsole.Enums;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies.Implementations;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies.Models;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies.PolicyRequirements;
+using Bit.Core.AdminConsole.Repositories;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.Policies.PolicyValidators;
+
+public abstract class OrganizationPolicyValidator(IPolicyRepository policyRepository, IEnumerable<IPolicyRequirementFactory<IPolicyRequirement>> factories) : IPolicyValidator
+{
+    private readonly IEnumerable<IPolicyRequirementFactory<IPolicyRequirement>> _factories = factories;
+
+    public abstract PolicyType Type { get; }
+
+    public abstract IEnumerable<PolicyType> RequiredPolicies { get; }
+
+
+    protected async Task<IEnumerable<T>> GetUserPolicyRequirementsByOrganizationIdAsync<T>(Guid organizationId, PolicyType policyType) where T : IPolicyRequirement
+    {
+        var policyDetails = await policyRepository.GetPolicyDetailsByOrganizationIdAsync(organizationId, policyType);
+        var policyDetailGroups = policyDetails.GroupBy(policyDetail => policyDetail.UserId);
```

Comment:
@eliykat 

I really like this logic. From the consumer's perspective, they're just getting a list of IPolicyRequirement. The IPolicyRequirement objects will be the same regardless of whether they came from here or `PolicyRequirementQuery`.

It also allows the `IPolicyRequirement` implementations to contain the logic for determining which users the policy should apply to.

Finally, we only need the `UserId` here and not elsewhere, so the `OrganizationPolicyDetails` object can handle that without requiring changes to existing code.

Please review and let me know what you think of this pattern. I'll add tests and clean up some comments once we're aligned on the big picture.